### PR TITLE
Add PWA support, Service Worker and Manifest, plus icons.

### PR DESCRIPTION
**Pull request by dalbitresb12**
Already talked with Evgeni via email about making QiFi a Progressive Web App. I added a Service Worker, Manifest file and icons needed by the Manifest.

### CODE REVIEW

**Comment by evgeni** [comment]
Thanks a ton! I'll have a closer look next week when I have some spare time, but this looks pretty polished already!

**Comment by dalbitresb12** [comment]
@evgeni, have you had time to review the pull request?

File: manifest.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "QiFi - WiFi QR Code Generator",
+  "short_name": "QiFI",
```
Comment (evgeni) [inline manifest.json]: QiFi, small I ;)

File: index.html [hunk 2]
```diff
@@ -23,10 +25,21 @@
...
+        if ('serviceWorker' in navigator) {
+            window.addEventListener('load', function() {
+                navigator.serviceWorker.register('/sw.js', {scope: './'}).then(function(registration) {
+                    console.log('SW Success')
+                }).catch(function(error) {
+                    console.log('SW failed:', error)
+                })
+            });
+        } else {
+            console.log('SW not supported')
```
Comment (evgeni) [inline index.html]: I don't think we need to spam the log in that case

File: sw.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  '/qifi.png',
+  '/qifi-small.png',
+  '/manifest.json',
+  '/jquery/jquery-3.3.1.min.js',
+  '/bootstrap/dist/js/bootstrap.min.js',
+  '/jquery-qrcode/jquery.qrcode.min.js',
+  '/jquery.storage.js/jquery.storage.js',
+  '/print.css',
+  '/bootstrap/dist/fonts/glyphicons-halflings-regular.woff2',
+  'https://button.flattr.com/flattr-badge-large.png'
```
Comment (evgeni) [inline sw.js]: This is gone now

**Review by evgeni — CHANGES_REQUESTED** [review]
As a matter of fact I did, it didn't work and I totally forgot to contact you.

I've retried now and it works like a charm. Whatever happened last time. 🤷

Posted a few minor comments, would you like to address them, or shall I?

**Comment by dalbitresb12** [comment]
This pull request is using an older version of QiFi, should I do this again with the latest version available on GitHub?

**Comment by evgeni** [comment]
Looking at the code, you only reference file names, those didn't change. So after merging it will automatically use the latest code.

**Comment by evgeni** [comment]
Btw, you can see a deployed version here: https://deploy-preview-18--qifi.netlify.com/

**Comment by dalbitresb12** [comment]
Is it okay now, @evgeni?
I'm kind of a newbie using Git, that's why I did two commits without noticing.

**Comment by dalbitresb12** [comment]
I've checked it and it's totally wrong, probably because I don't know how to use Git. I'll try to fix this. So sorry :(

**Comment by dalbitresb12** [comment]
I think this is now fixed. Sorry, I'm a total newbie using Git.

This still has conflicts in the index.html because I'm using an older commit.

**Comment by dalbitresb12** [comment]
@evgeni, have you checked the changes?

**Comment by evgeni** [comment]
Yes, I did! Merging :)

**Pull request merged**